pygmt/src/rose.py

```diff
@@ -0,0 +1,245 @@
+          attributes. Finally, to instead save the computed mean
+          direction and other statistics, use
+          ``vectors = '+wmode_file'``. The eight items saved to
+          a single record are: *mean_az*, *mean_r*, *mean_resultant*,
+          *max_r*, *scaled_mean_r*, *length_sum*, *n*, *sign@alpha*,
+          where the last term is 0 or 1 depending on whether the mean
+          resultant is significant at the level of confidence set via
+          **conf_level**.
+
+    vector_params : str
+        Used with **vectors** to modify vector parameters. For
+        vector heads, append vector head size [Default is 0, i.e., a
+        line]. See :gmt-docs:`rose.html#vector-attributes` for
+        specifying additional attributes. If **vectors** is not
+        given and the current plot mode is to draw a windrose diagram
+        then using **vector_params** will add vector heads to all
+        individual directions using the supplied attributes.
+
+    alpha : float or str
+        Sets the confidence level used to determine if the mean
+        resultant is significant (i.e., Lord Rayleigh test for
+        uniformity) [``alpha = 0.05``]. Note: The critical
+        values are approximated [Berens, 2009] and requires at
+        least 10 points; the critical resultants are accurate to at
+        least 3 significant digits. For smaller data sets you
+        should consult exact statistical tables.
+
+        Berens, P., 2009, CircStat: A MATLAB Toolbox for Circular
+        Statistics, *J. Stat. Software*, 31(10), 1-21,
+        https://doi.org/10.18637/jss.v031.i10.
+
+    inquire : bool
+        Inquire. Computes statistics needed to specify a useful
+        **frame**. No plot is generated. The following statistics
+        are written to stdout: *n*, *mean az*, *mean r*, *mean
+        resultant length*, *max bin sum*, *scaled mean*, and
+        *linear length sum*.
+
+    {U}
+    {V}
+    {XY}
+    {p}
+    {t}
+    """
+
+    kwargs = self._preprocess(**kwargs)
+
+    kind = data_kind(data, length, azimuth)
+
+    extra_arrays = []
+    if "t" in kwargs and is_nonstr_iter(kwargs["t"]):
+        extra_arrays.append(kwargs["t"])
+        kwargs["t"] = ""
+
+    with Session() as lib:
+        # Choose how data will be passed in to the module
+        if kind == "file":
+            file_context = dummy_context(data)
+        elif kind == "matrix":
+            file_context = lib.virtualfile_from_matrix(data)
+        elif kind == "vectors":
+            file_context = lib.virtualfile_from_vectors(
+                np.atleast_1d(length), np.atleast_1d(azimuth), *extra_arrays
+            )
+
+        with file_context as fname:
+            # if JX aka diameter is not given add it with default of 7.5c
+            if "JX" not in kwargs:
+                arg_str = " ".join([fname, build_arg_string(kwargs), "-JX7.5c"])
+            else:
+                arg_str = " ".join([fname, build_arg_string(kwargs)])
```

Comment:
Related to an upstream issue: https://github.com/GenericMappingTools/gmt/issues/4722

BTW, it's a bad idea to set the diameter to 7.5 cm. 

For example:
```
fig = pygmt.Figure()
fig.rose(data="@fractures_06.txt", region=[0, 1, 0, 360], ...)
fig.rose(data="@fractures_06.txt", region=[0, 1, 0, 360], projection="X10c", ...)
fig.rose(data="@fractures_06.txt", region=[0, 1, 0, 360], ...)
fig.rose(...)
```
1. the first rose diameter is 7.5 cm, because`-JX` is not given. 
2. the second rose diameter is 10 cm, due to the `projection` argument
3. the third rose diameter is 7.5 cm, because `-JX` is not given. However, the expected rose diameter is 10 cm, because the third rose diamter should *remember* the projection of the previous command if `JX` is not given.

Reply:
The issue was fixed in upstream GMT. Although it won't work for GMT v6.1.1, I don't think it's a big issue. So we can simplify the codes to:
```suggestion
            arg_str = " ".join([fname, build_arg_string(kwargs)])
```

Reply:
Thanks for pushing that issue forward @seisman .